src/buffer/out/TextAttribute.cpp

```diff
@@ -104,15 +104,34 @@ std::pair<COLORREF, COLORREF> TextAttribute::CalculateRgbColors(const std::array
                                                                 const COLORREF defaultBgColor,
                                                                 const bool reverseScreenMode,
                                                                 const bool blinkingIsFaint,
-                                                                const bool boldIsBright) const noexcept
+                                                                const bool boldIsBright,
+                                                                const std::optional<std::array<std::array<COLORREF, 18>, 18>>& adjustedForegroundColors) const noexcept
 {
-    auto fg = _foreground.GetColor(colorTable, defaultFgColor, boldIsBright && IsBold());
+    COLORREF fg;
     auto bg = _background.GetColor(colorTable, defaultBgColor);
+    bool reversed{ false };
+    if (adjustedForegroundColors.has_value() &&
+        (_background.IsDefault() || _background.IsLegacy()) &&
+        (_foreground.IsDefault() || _foreground.IsLegacy()))
+    {
+        auto bgIndex = _background.IsDefault() ? 16 : _background.GetIndex();
+        auto fgIndex = _foreground.IsDefault() ? 17 : _foreground.GetIndex();

```

Comment:
ugh magic numbers

Reply:
Wouldn't it be possible to calculate the default foreground/background colors eagerly and stuff them into the `_defaultFg` and `_defaultBg` members?

Reply:
IsDefault does not mean "are these the same as the default values"; it means "are these the unspecified SGR 38 and SGR 48 colors indicating _do what the user says_."